src/main/java/de/retest/recheck/persistence/bin/KryoPersistence.java

```diff
@@ -102,12 +103,35 @@ public void save( final URI identifier, final T element ) throws IOException {
 		}
 	}
 
-	@SuppressWarnings( "unchecked" )
+	/**
+	 * You need to maintain the stream and are responsible for closing it properly.
+	 */
+	public void save( final OutputStream outputStream, final T element ) throws IOException {
+		final Output output = new Output( new LZ4FrameOutputStream( outputStream ) );
+		output.writeString( version );
+		kryo.writeClassAndObject( output, element );
+		// Flush the stream to avoid corrupt file data
+		output.flush();
+	}
+
 	@Override
 	public T load( final URI identifier ) throws IOException {
 		final Path path = Paths.get( identifier );
+		try ( InputStream input = newInputStream( path ) ) {
+			return load( input, identifier );
+		} catch ( final IncompatibleReportVersionException | NoSuchFileException e ) {
+			throw e;
+		}
+	}
+
+	/**
+	 * You need to maintain the stream and are responsible for closing it properly.
+	 */
+	@SuppressWarnings( "unchecked" )
+	public T load( final InputStream in, final URI identifier ) throws IOException {
 		String writerVersion = null;
-		try ( final Input input = new Input( new LZ4FrameInputStream( newInputStream( path ) ) ) ) {
+		try {
+			final Input input = new Input( new LZ4FrameInputStream( in ) );
```

Comment:
Why did you change this? This was fine as it is.

Reply:
No it was not. We _must_ close the outermost stream, not some stream "in between". Closing here would have done the latter...